lib/functional.fnl

```diff
@@ -0,0 +1,209 @@
+(local fu hs.fnutils)
```

Comment:
Although functions in this namespace are succinct and easy to reason about, it would be nice to add a dosctring to each. Perhaps we can steal some from clojure.core source

Reply:
Done in #36